include/tbbexec/tbb_thread_pool.hpp

```diff
@@ -0,0 +1,413 @@
+            }
+
+            template <stdexec::__decays_to<bulk_sender> Self, class Env>
+            friend auto tag_invoke(stdexec::get_completion_signatures_t, Self&&, Env)
+                -> stdexec::dependent_completion_signatures<Env>;
+
+            template <stdexec::__decays_to<bulk_sender> Self, class Env>
+            friend auto tag_invoke(stdexec::get_completion_signatures_t, Self&&, Env)
+                -> completion_signatures<Self, Env>
+                requires true
+            ;
+
+            template <stdexec::tag_category<stdexec::forwarding_sender_query> Tag, class... As>
+                requires stdexec::__callable<Tag, const Sender&, As...>
+            friend auto tag_invoke(Tag tag,
+                                   const bulk_sender& self,
+                                   As&&... as) noexcept(stdexec::__nothrow_callable<Tag, const Sender&, As...>)
+                -> stdexec::__call_result_if_t<stdexec::tag_category<Tag, stdexec::forwarding_sender_query>,
+                                               Tag,
+                                               const Sender&,
+                                               As...> {
+                return ((Tag &&) tag)(self.sndr_, (As &&) as...);
+            }
+        };
+
+        friend sender tag_invoke(stdexec::schedule_t, const scheduler& s) noexcept { return sender{*s.pool_}; }
+
+        template <stdexec::sender Sender, std::integral Shape, class Fun>
+        using bulk_sender_t =
+            bulk_sender<stdexec::__x<std::remove_cvref_t<Sender>>, Shape, stdexec::__x<std::remove_cvref_t<Fun>>>;
+
+        template <stdexec::sender S, std::integral Shape, class Fn>
+        friend bulk_sender_t<S, Shape, Fn> tag_invoke(stdexec::bulk_t,
+                                                      const scheduler& sch,
+                                                      S&& sndr,
+                                                      Shape shape,
+                                                      Fn fun) noexcept {
+            return bulk_sender_t<S, Shape, Fn>{*sch.pool_, (S &&) sndr, shape, (Fn &&) fun};
+        }
+
+        friend stdexec::forward_progress_guarantee tag_invoke(stdexec::get_forward_progress_guarantee_t,
+                                                              const tbb_thread_pool&) noexcept {
+            return stdexec::forward_progress_guarantee::parallel;
+        }
+
+        friend class tbb_thread_pool;
+        explicit scheduler(tbb_thread_pool& pool) noexcept : pool_(&pool) {}
+
+        tbb_thread_pool* pool_;
+    };
+
+    scheduler get_scheduler() noexcept { return scheduler{*this}; }
+
+    void request_stop() noexcept { /* TODO? */
+    }
+
+    std::uint32_t available_parallelism() const { return m_arena.max_concurrency(); }
+
+private:
+    void enqueue(task_base* task) noexcept {
+        m_arena.enqueue([task] { tbb::this_task_arena::isolate([task] { task->__execute(task, /*tid=*/0); }); });
```

Comment:
I added `tbb::this_task_arena::isolate` to prevent [moonlighting](https://link.springer.com/chapter/10.1007/978-1-4842-4398-5_12). Not sure if it's necessary, but I feel like in the context of P2300, there's no benefit of moonlighting and plenty of footguns.

Reply:
Please see comment on `bulk_enqueue` below.  Does P2300 have things to say about ungood things like messing with `thread_local`, for work not in `bulk`?

Reply:
Would you also consider removing `isolate` here?

Reply:
Sure thing. The [one library feature I wrote for TBB](https://github.com/oneapi-src/oneTBB/issues/267) got me paranoid, but that was `tbb::collabrative_call_once` which has the potential to deadlock without isolation.

What should go in the `tid` field of `__execute`? It's just
```c++
  struct task_base {
    task_base* next;
    void (*__execute)(task_base*, std::uint32_t tid) noexcept;
  };
```
without comment.

Reply:
@BenFrantzDale Thanks for asking!  The last time I used TBB seriously was in 2010, so I'll have to look carefully at the documentation to answer this.  (Eric pulled me in to review because I was the only one available on the spot who knew _anything_ about TBB ; - ) .)  I did have to subclass `task`, but that was already considered expert-only back then.  I can only guess that "tid" suggests the native thread id.